app/views/dashboards/_budget_bar.html.erb

```diff
@@ -16,13 +16,28 @@
 
 <div class="row" id='budget-report'>
   <% expenditures.each_pair do |type, patient_hashes| %>
-    <div class="col-sm-4">
-      <%= t("dashboard.budget_bar.#{type}_report",
-            amount: number_to_currency(expenditures[type].map { |h| h[:fund_pledge] }.inject(:+) || 0,
-                                       precision: 0,
-                                       unit: '$',
-                                       format: '%u%n'),
-            count: expenditures[type].count) %>
+    <div class="col-sm-3">
+      <%= budget_bar_statistic(t("dashboard.budget_bar.#{type}"),
+                               expenditures[type],
+                               limit: limit,
+                               show_aggregate_statistics: Config.show_aggregate_statistics?) %>
+    </div>
+  <% end %>
+
+  <% if Config.show_aggregate_statistics? %>
```

Comment:
checking the config twice here. looks like it gets cached, but is there a better way? perhaps more generally, does this architecture (statistic-string builder + options) seem ok to you?

Reply:
yeah, this is a very railsy pattern for html generation and I think it serves us nicely here. I don't think double-checking is problematic here (as you noted, query gets cached) but the way around this for a heavier query that didn't get cached would be to declare this as an instance variable in the controller, e.g. `@show_stats = Config.show_aggregate_statistics?` and then check that variable directly. (Again, to be clear, shouldn't do that here thanks to db cache magic!)